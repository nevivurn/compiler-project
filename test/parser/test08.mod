//
// test08
//
// parser test:
// - strings
// - built-in routines WriteStr() and WriteLn()
//

module test08;

begin
  WriteStr("Hello, world!"); WriteLn()
end test08.
