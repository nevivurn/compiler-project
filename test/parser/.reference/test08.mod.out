parsing 'test08.mod'...
successfully parsed.
  AST:
    CAstScope: 'test08'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @_str_1   array 14 of <char>>      _str_1           ]
            [ data: "Hello, world!" ]
          [ main     <NULL>           ]
        ]]
      statement list:
        call [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL> ] <NULL>
          "Hello, world!" array 14 of <char>>
        call [ *WriteLn() --> <NULL> ] <NULL>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest08.mod.ast.pdf test08.mod.ast.dot


Done.
